pkg/workos_errors/authentication_errors.go

```diff
@@ -0,0 +1,204 @@
+package workos_errors
+
+import (
+	"errors"
+)
+
+// Authentication error code constants
+const (
+	EmailVerificationRequiredCode                 = "email_verification_required"
+	MFAEnrollmentCode                             = "mfa_enrollment"
+	MFAChallengeCode                              = "mfa_challenge"
+	OrganizationSelectionRequiredCode             = "organization_selection_required"
+	SSORequiredCode                               = "sso_required"
+	OrganizationAuthenticationMethodsRequiredCode = "organization_authentication_methods_required"
+)
+
+// AuthenticationFactor represents an MFA factor
+type AuthenticationFactor struct {
+	ID   string `json:"id"`
+	Type string `json:"type"`
+}
+
+// Organization represents an organization in selection error
+type Organization struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
```

Comment:
Just noting here that this type is separate in the backend, so we may want to keep this one separate. Small naming recommendation to delineate from other Organization struct:

```suggestion
// Organization represents an organization in selection error
type PendingAuthenticationOrganizationInfo struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}
```

Reply:
Updated to use the PendingAuthenticationOrganizationInfo naming convention